### ISSUE TITLE
RuntimeOptions are not taken into account for onWrite / onCreate-trigger functions

### ISSUE THREAD

**Issue by igorgarbuz** [firebase/firebase-tools#5945]
### [REQUIRED] Environment info

**firebase-tools: 11.26.0**

**Platform:**
MacOS and Ubuntu, Node 16.

### [REQUIRED] Test case

Define function in a file as following and export it in `src/index.ts`.  
```
import { RuntimeOptions } from 'firebase-functions';
import * as functions from 'firebase-functions';

const functionsInRegion = functions.region('europe-west1');

const runtimeOptions: RuntimeOptions = {
    memory: '512MB',
    timeoutSeconds: 240,
};

const handleCreate = async (
    createdDocSnapshot: FirebaseFirestore.DocumentSnapshot,
    context: functions.EventContext,
) => {
    const { parameter } = context.params;

    // Actions with `parameter` here
};

export default functionsInRegion
    .runWith(runtimeOptions)
    .firestore.document(
        `collectionName/{documentId}`,
    )
    .onCreate(handleCreate);

```
Deploy finished without errors and function works properly, except the timeout time is not taken into account. For some large documents function may take longer than 60 sec and be terminated because of timeout timer, which is visible in function logs.

### [REQUIRED] Steps to reproduce

1. Create a firebase function with `onWrite` or `onCreate` trigger.
2. Define runtimeOptions like:
```
const runtimeOptions: RuntimeOptions = {
    memory: '512MB',
    timeoutSeconds: 240,
};
```
3. Pass runtimeOptions to a function with a method `.runWith(runtimeOptions)`
4. Deploy the function:
```
firebase deploy --only functions:<functionName>
```
5. Check GCP functions dashboard, just deployed function has default 256MB and 60s options instead of specified `runtimeOptions`.

7. Modify these parameters by hand in GCP Functions dashbaord UI, submit and redeploy function through the UI. Manually set parameters are taken into account.

### [REQUIRED] Expected behavior

Deployed function has 512MB of memory and 240s of timeout when deploying with firebase-tools CLI.

### [REQUIRED] Actual behavior

Default runtime options are set. Specified runtime options are not taken into account.

**Comment by Berlioz** [comment]
I've been trying to replicate this and not having any luck.  On OSX 13.4, with Node 16, with firebase-tools 11.26 and 12.3.1 (latest) and using that exact code sample, I keep creating functions which seem to have the appropriate memory and timeout values. The only suggestion I have off the top of my head is to try updating firebase-tools anyway and creating a new project directory, in case there's some kind of weird contamination with your local state? Sorry I can't be more helpful here.

![Screenshot 2023-06-09 at 2 17 19 PM](https://github.com/firebase/firebase-tools/assets/616364/c974b29a-7150-40b6-b851-088722515d18)

**Comment by igorgarbuz** [comment]
@Berlioz thank you for trying to reproduce.
We tried to update the firebase-tools. Also the arguments are not taken into account when deploying from many different systems : MacOS for myself, Ubuntu for somme team-mates, debian for the CI. I've also tried to re-install the project, but still no luck.

We don't have access to some system logs that would allow to see what causing this.
Maybe we can give you an access to our firebase project to try reproducing it there ?
